### PR TITLE
Fix/wcn 196 eddsa recipient guard

### PR DESCRIPTION
**Pull request by mrdanish26**
## Summary

  Follow-up to WAL-375 (#8539). Applies the same recipient verification guard to the **EdDSA** TSS signing path.

  ### The vulnerability (same root cause as WAL-375)

  `EddsaUtils.signRequestBase()` did not call `verifyTransaction` with the actual recipients, allowing a compromised API server to substitute a malicious `signableHex` — the transaction presented to the user for signing could differ from what was actually broadcast.

  ### Fix

  Added `resolveEffectiveTxParams` + `verifyTransaction` call inside `signRequestBase()` for `RequestType.tx`, immediately after `unsignedTx` is resolved and before key derivation begins:

  ```typescript
  const effectiveTxParams = resolveEffectiveTxParams(txRequestResolved, params.txParams);
  await this.baseCoin.verifyTransaction({
    txPrebuild: { txHex: unsignedTx.signableHex },
    txParams: effectiveTxParams,
    wallet: this.wallet,
    walletType: this.wallet.multisigType(),
  });

  EdDSA-specific NO_RECIPIENT_TX_TYPES expansion

  EdDSA coins have many more transaction types that legitimately carry no recipients. The shared NO_RECIPIENT_TX_TYPES set in recipientUtils.ts was expanded from 8 → 44 types:

  ┌───────────────────────────────────────────────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
  │                   Category                    │                                                                                                          Types                                                                                                           │
  ├───────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Staking (SOL, ADA, NEAR, DOT, CSPR, SUI, APT) │ stakingActivate, stakingDeactivate, stakingWithdraw, stakingClaim, stakingDelegate, stakingUnlock, stakingUnvote, stakingPledge, stakingAuthorize, stakingAuthorizeRaw, stakingLock, stakingAdd, addStake, withdrawStake │
  ├───────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Account / wallet init                         │ walletInitialization, addressInitialization, createAccount, accountUpdate                                                                                                                                                │
  ├───────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Token operations                              │ trustline, closeAssociatedTokenAccount, associatedTokenAccountInitialization, tokenTransfer, sendToken, sendNFT                                                                                                          │
  ├───────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ NEAR                                          │ storageDeposit                                                                                                                                                                                                           │
  ├───────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ TON                                           │ singleNominatorWithdraw, tonWhalesDeposit, tonWhalesWithdrawal, tonWhalesVestingDeposit, tonWhalesVestingWithdrawal                                                                                                      │
  ├───────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ADA                                           │ voteDelegation                                                                                                                                                                                                           │
  ├───────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Canton 2-step transfers                       │ transferAccept, transferReject, transferAcknowledge, transferOfferWithdrawn, oneStepPreApproval                                                                                                                          │
  └───────────────────────────────────────────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

  intent.intentType fallback

  EdDSA coins populate txRequest.intent.intentType but often don't set txParams.type. The guard now falls back to intent.intentType when txParams.type is absent — this is required for EdDSA staking and wallet init flows to pass the guard without breaking.

  Test plan

  - recipientUtils unit tests — 20 passing (ECDSA types, EdDSA staking types, Canton types, intentType fallback)
  - EdDSA signTxRequest tests — 9 passing (rejection for payment without recipients, acceptance for stakingActivate and walletInitialization)
  - ECDSA MPCv2 signTxRequest tests — 13 passing (no regression on shared recipientUtils.ts)

  Known edge cases

  - XLM: verifyTransaction makes 2 keychain API calls per sign — ~100–300ms latency, pre-existing behaviour not introduced here
  - HBAR / DOT: strict recipient checks in verifyTransaction for some types — mitigated by comprehensive exempt list; raise a follow-up if new types surface
  ```

TICKET: WCN-196

### CODE REVIEW

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts [hunk 1]
```diff
@@ -610,6 +611,14 @@ export class EddsaUtils extends baseTSSUtils<KeyShare> {
       );
       unsignedTx =
```
Comment (pranishnepal) [inline modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts:612]: why do we only consider the first item in the array - is that the only one that _matters_? 
```
 txRequestResolved.transactions![0].unsignedTx : txRequestResolved.unsignedTxs[0];
```

File: modules/sdk-core/test/unit/bitgo/utils/tss/recipientUtils.ts [hunk 2]
```diff
@@ -98,7 +122,22 @@ describe('recipientUtils', function () {
...
 
-    const NO_RECIPIENT_TYPES = [
+    it('allows empty recipients when txParams.type is a no-recipient type', function () {
+      const { resolveEffectiveTxParams } = getModule();
+      const txRequest = makeTxRequest();
+      const result = resolveEffectiveTxParams(txRequest, { type: 'stakingActivate' });
+      result.type.should.equal('stakingActivate');
+    });
+
+    it('allows empty recipients when intent.intentType is a no-recipient type (EdDSA fallback)', function () {
```
Comment (pranishnepal) [inline modules/sdk-core/test/unit/bitgo/utils/tss/recipientUtils.ts:132]: maybe we can add a test to also validate that the request is rejected when intentType is some unknown type? Or, is that already covered?

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts (hunk 1 above)
Comment (mrdanish26) [reply modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts:612]: Each `signTxRequest` call handles a single transaction, the array structure exists in the type for historical/batch reasons, but in practice only index 0 is ever populated per signing round. The same pattern is used identically in the `ECDSA` path (ecdsa.ts and ecdsaMPCv2.ts)

File: modules/sdk-core/test/unit/bitgo/utils/tss/recipientUtils.ts (hunk 2 above)
Comment (mrdanish26) [reply modules/sdk-core/test/unit/bitgo/utils/tss/recipientUtils.ts:132]: This is already covered. The test `throws when txParams is undefined and intent has no recipients` uses intentType: 'payment' with no txParams. Since payment isn't in the exempt list, the guard throws. That's the rejection path for an unknown intent type.

**Review by pranishnepal — APPROVED** [review]

**Review by zahin-mohammad — COMMENTED** [review]
in general looks good, but please also get a review from @mohammadalfaiyazbitgo
